src/Common/tests/System/AssertExtensions.cs

```diff
@@ -13,5 +14,17 @@ public static void Throws<T>(Action action, string message)
         {
             Assert.Equal(Assert.Throws<T>(action).Message, message);
         }
+
+        [Conditional("DEBUG")]
+        public static void ThrowsInDebug<T>(Action action) where T : Exception => Assert.Throws<T>(action);
+
+        [Conditional("DEBUG")]
+        public static void ThrowsInDebug<T>(Func<object> func) where T : Exception => Assert.Throws<T>(func);
+
+        [Conditional("DEBUG")]
+        public static void ThrowsAnyInDebug<T>(Action action) where T : Exception => Assert.ThrowsAny<T>(action);
+
+        [Conditional("DEBUG")]
+        public static void ThrowsAnyInDebug<T>(Func<object> func) where T : Exception => Assert.ThrowsAny<T>(func);
```

Comment:
@stephentoub It's going to work in this case since the Common file is #included in the same assembly as the tests.